### PR TITLE
[Fix] re-design fix inventory search

### PR DESCRIPTION
**Pull request by sijav**
# Description
re-design fix inventory search

# To-Dos

- [x] Extract i18n strings via `yarn i18n:extract`
- [x] Check formatting via `yarn format:check`
- [x] Lint via `yarn lint` and `yarn lint:tsc`
- [x] Test via `yarn test`

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://some.engineering/code-of-conduct).

### CODE REVIEW

File: src/shared/fix-query-parser/query.ts [hunk 1]
```diff
@@ -925,12 +949,87 @@ export class Query {
...
+        if (item) {
+          item.text = value
+        } else {
+          draft.working_part.term = draft.working_part.term.set_last(new FulltextTerm({ text: value }), 'and')
+        }
+      } else if (item) {
+        draft.working_part.term = draft.working_part.term.delete_terms((t) => t === item)
+      }
+    })
+  }
```
Comment (aquamatthias) [inline src/shared/fix-query-parser/query.ts:987]: why would we ever setting by index? I think this does not make sense.

File: src/shared/fix-query-parser/query.ts [hunk 2]
```diff
@@ -925,12 +949,87 @@ export class Query {
...
+
+  public delete_fulltext_index(index: number): Query {
+    return produce(this, (draft) => {
+      const item = draft.fulltextPredicates()[index]
+      if (item) {
+        draft.working_part.term = draft.working_part.term.delete_terms((t) => t === item)
+      }
+    })
+  }
```
Comment (aquamatthias) [inline src/shared/fix-query-parser/query.ts:1016]: remove

File: src/shared/fix-query-parser/query.ts [hunk 3]
```diff
@@ -925,12 +949,87 @@ export class Query {
...
+
+  public delete_fulltext(value: string): Query {
+    return produce(this, (draft) => {
+      const item = draft.fulltextPredicates().find((i) => i.text === value)
+      if (item) {
+        draft.working_part.term = draft.working_part.term.delete_terms((t) => t === item)
+      }
+    })
+  }
```
Comment (aquamatthias) [inline src/shared/fix-query-parser/query.ts:1025]: consider to rename set to update - it is already able to delete

File: src/shared/fix-query-parser/query.ts [hunk 4]
```diff
@@ -959,6 +1058,15 @@ export class Query {
...
 
+  public delete_all_term_if_any(): Query {
+    const existing = this.working_part.term.find_terms((i) => i instanceof AllTerm)
+    return existing.length > 0
+      ? produce(this, (draft) => {
+          draft.working_part.term = draft.working_part.term.delete_terms((t) => t instanceof AllTerm)
+        })
+      : this
+  }
+
```
Comment (aquamatthias) [inline src/shared/fix-query-parser/query.ts:1069]: can be removed

**Review by aquamatthias — CHANGES_REQUESTED** [review]

File: src/shared/fix-query-parser/query.ts (hunk 4 above)
Comment (sijav) [reply src/shared/fix-query-parser/query.ts:1016]: No This is to remove AllTerm when there's one. this solution is for when we're trying to add a term to a query when there's a all in it

File: src/shared/fix-query-parser/query.ts [hunk 5]
```diff
@@ -276,6 +283,13 @@ export abstract class Term {
     return this
   }
 
+  set_last(term: Term, op: string): Term {
```
[the remark this answers is not in the document]
Comment (sijav) [reply src/shared/fix-query-parser/query.ts]: removed

**Review by aquamatthias — DISMISSED** [review]
LGTM - Thanks

**Review by aquamatthias — APPROVED** [review]

**Pull request merged**